src/kemal/config.cr

```diff
@@ -88,14 +89,26 @@ module Kemal
       FILTER_HANDLERS << handler
     end
 
+    # Returns the defined error handlers for HTTP status codes
     def error_handlers
       ERROR_HANDLERS
     end
 
+    # Adds an error handler for the given HTTP status code
     def add_error_handler(status_code : Int32, &handler : HTTP::Server::Context, Exception -> _)
       ERROR_HANDLERS[status_code] = ->(context : HTTP::Server::Context, error : Exception) { handler.call(context, error).to_s }
     end
 
+    # Returns tje defined error handlers for exceptions
+    def exception_handlers
+      EXCEPTION_HANDLERS
+    end
+
+    # Adds an error handler for the given exception
+    def add_exception_handler(exception : Exception.class, &handler : HTTP::Server::Context, Exception -> _)
+      EXCEPTION_HANDLERS[exception] = ->(context : HTTP::Server::Context, error : Exception) { handler.call(context, error).to_s }
```

Comment:
I believe this can be added as given, avoiding an additional closure.

```suggestion
      EXCEPTION_HANDLERS[exception] = handler
```

Reply:
Yeah I agree requiring `String` output type should be preferred in general. This might have some UX consequences though...
@syeopite What's the motivation for an unspecified block output type?

Reply:
No reason in particular. It's just copied from the original `add_error_handler` but forcing an specified block type does seem like a breaking change to me. 

Take this error handler defined in the tests for example:

https://github.com/kemalcr/kemal/blob/749c537e853af6f032b01cf2b91ae2e740340d62/spec/exception_handler_spec.cr#L66-L68

It has an output type of `String | Nil` instead of just a `String` due to it returning the message attribute of the exception.

Some end-user applications might also use similar patterns that end up an output type that may or may not be a `String`.